src/com/minseung/datastructure/LinkedListNode.java

```diff
@@ -225,37 +235,38 @@ public static void main(String[] args) {
         ll.append(3);
         ll.append(4);
         ll.append(5);
-        ll.printData(ll);
+        ll.printData();
         //found = ll.findKthNode(ll.getFirst(), k, r);
-        found = ll.findKthNodeBasic(ll.getFirst(), k);
+        found = ll.findKthNodeBasic(ll.getHeader(), k);
         System.out.println(found.data);
 
-        found = ll.findKthNodeUsingRecursive(ll.getFirst(), k, r);
+        found = ll.findKthNodeUsingRecursive(ll.getHeader(), k, r);
         System.out.println(found.data);
 
-        found = ll.findKthNodeUsingPointer(ll.getFirst(), k);
+        found = ll.findKthNodeUsingPointer(ll.getHeader(), k);
         System.out.println(found.data);
         ll.deleteMiddleNode(ll.getNode(k));
-        ll.printData(ll);
-        /*
+        ll.printData();
+
         LinkedList l1 = new LinkedList();
         LinkedList l2 = new LinkedList();
         l1.append(9);
         l1.append(1);
         l1.append(4);
-        l1.retrieve();
+        l1.printData();
         l2.append(6);
         l2.append(4);
-        l2.append(3);
-        l2.retrieve();
+        l2.append(9);
+        //l2.append(9);
+        l2.printData();
         LinkedListNode linkedListNode = new LinkedListNode();
-        LinkedList l3 = linkedListNode.getSum(l1, l2);
-        l3.retrieve();
-        */
+        LinkedList l3 = linkedListNode.getSumUsingIterate(l1, l2);
+        l3.printData();
+
 
     }
 
-    public LinkedList getSum(LinkedList l1, LinkedList l2) {
+    public LinkedList getSumUsingIterate(LinkedList l1, LinkedList l2) {
```

Comment:
Or, you can create a child class from this class, and implement getSum differently.